syntax = "proto3";

package accountmanagement;

message AccountRequest {
    string name = 1;
    string email = 2;
    string password = 3;
}

message AccountUpdateRequest {
    string id = 1;
    string name = 2;
    string email = 3;
    string password = 4;
}

message AccountId {
    string id = 1;
}

message AccountResponse {
    string id = 1;
    string name = 2;
    string email = 3;
}

message DeleteResponse {
    string message = 1;
}

message AccountEmail {
    string email = 1;
}

service AccountService {
    rpc CreateAccount (AccountRequest) returns (AccountResponse);
    rpc GetAccount (AccountId) returns (AccountResponse);
    rpc UpdateAccount (AccountUpdateRequest) returns (AccountResponse);
    rpc GetAccountByEmail (AccountEmail) returns (AccountResponse);
    rpc DeleteAccount (AccountId) returns (DeleteResponse);
}